File: 169MajorityElement.go
package main

import "fmt"

func main() {
	nums := []int{6, 5, 5}
	fmt.Println(majorityElement(nums))
}

func majorityElement(nums []int) int {
	m := map[int]int{}
	for _, num := range nums {
		m[num] = m[num] + 1
	}
	fmt.Println(m)
	for num, count := range m {
		if float64(count) >= float64(len(nums))/2.0 {
			return num
		}
	}
	return -1
}

File: 169MajorityElement.java
import java.util.*;

class MajorityElement {
  public static void main(String[] args) {
    int[] nums = {3,2,3};
    
    System.out.print(majorityElement(nums));
  }

  public static int majorityElement(int[] nums) {
    HashMap<Integer, Integer> count = new HashMap<>();
    for(int i : nums){
      if(!count.keySet().contains(i)){
        count.put(i, 0);
      }
      count.put(i, count.get(i)+1);
    }

    int half = nums.length/2;
    int output = 0;
    for(int j : count.keySet()){
      if(count.get(j) > half){
        output = j;
      }
    }
    return output;
  }
}